lib/Target/TTMetal/TTMetalToFlatbuffer.cpp

```diff
@@ -252,17 +255,16 @@ static std::shared_ptr<void> translateModuleToFlatbuffer(Operation *op) {
             emitDispatchOpRegionsAsCpp(dispatchOp, cppKernels);
         assert(success.succeeded() &&
                "failed to emit dispatch op regions as cpp");
-
         for (auto &region : dispatchOp.getRegions()) {
           std::vector<::tt::target::Dim2dRange> coreRangeSet = {
               toFlatbuffer(mlir::cast<CoreRangeAttr>(
                   dispatchOp.getCoreRanges()[region.getRegionNumber()]))};
           std::vector<::flatbuffers::Offset<::tt::target::CBRef>> cbs;
+          size_t argNumber = 0;
           for (auto arg : region.getArguments()) {
-            assert(arg.getArgNumber() < operands.size());
             auto cbType = mlir::cast<ttkernel::CBType>(arg.getType());
             auto cbDesc = cache.getOrCreate(cbType, cbTypeToFlatbuffer);
-            auto tensorRef = operands[arg.getArgNumber()];
+            auto tensorRef = cbType.getIsInternal() ? 0 : operands[argNumber++];
```

Comment:
Why do we do this?  Is this for the workaround you mentioned below where we were seeing weird data mismatch tiling issues?

Reply:
It's not a workaround for the mismatch. ScalerCB does not have a tensor operand reference, it's kinda internal to the kernel itself. So we skip creating tensor ref in the FB here using this attribute.